x-pack/solutions/security/plugins/security_solution/server/lib/detection_engine/rule_types/create_security_rule_type_wrapper.ts

```diff
@@ -567,7 +568,9 @@ export const createSecurityRuleTypeWrapper: CreateSecurityRuleTypeWrapper =
                   gapRange: experimentalFeatures.storeGapsInEventLogEnabled ? gap : undefined,
                   frozenIndicesQueriedCount,
                 },
-                userError: result.userError,
+                userError:
+                  result.userError ||
+                  result.errors.some((err) => checkErrorDetails(err).isUserError),
```

Comment:
we should only report `userError` for the overall status if all errors are user errors